packages/core/src/sdk/analytics/hooks/usePageViewEvent.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import { sendAnalyticsEvent } from '@faststore/sdk'
+import type { PageViewEvent } from '@faststore/sdk'
+
+export const usePageViewEvent = () => {
+  const sendPageViewEvent = useCallback(() => {
+    sendAnalyticsEvent<PageViewEvent>({
+      name: 'page_view',
+      params: {
+        page_title: document.title,
+        page_location: location.href,
+        send_page_view: true,
+      },
+    })
+  }, [])
+
+  const router = useRouter()
+
+  useEffect(() => {
+    router.events.on('routeChangeComplete', sendPageViewEvent)
```

Comment:
In my tests, it will trigger only when happen route change. So when the user enters a screen for the first time (Page initialization). The event is not triggered.